From 00e6bce512e3a34764a12ef65cc32ce1c29dc617 Mon Sep 17 00:00:00 2001
From: Evgeny Eltsin <eaeltsin@google.com>
Date: Tue, 20 Jul 2010 21:24:42 +0400
Subject: [PATCH 28/29] [gcc] Fix access to local symbols in 64-bit pic.

This fix changes movsi template for lea to print the first operand as %a1 instead of %1. Gcc trunk has this fix in, so this change looks like a fix for gcc bug, unrelated to nacl;

This fix also changes movsi template attr of TYPE_LEA to match operand as SImode instead of DImode. Gcc trunk still has DImode, but this seems to be a bug...

Review URL: http://codereview.chromium.org/2858059
---
 gcc/gcc/config/i386/i386.md |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/gcc/gcc/config/i386/i386.md b/gcc/gcc/config/i386/i386.md
index 79e75fa..2a1d573 100644
--- a/gcc/gcc/config/i386/i386.md
+++ b/gcc/gcc/config/i386/i386.md
@@ -1569,7 +1569,7 @@
       return "movd\t{%1, %0|%0, %1}";
 
     case TYPE_LEA:
-      return "lea{l}\t{%1, %0|%0, %1}";
+      return "lea{l}\t{%a1, %0|%0, %a1}";
 
     default:
       gcc_assert (!flag_pic || LEGITIMATE_PIC_OPERAND_P (operands[1]));
@@ -1585,7 +1585,7 @@
 	      (const_string "sselog1")
 	    (eq_attr "alternative" "7,8,9,10,11")
 	      (const_string "ssemov")
- 	    (match_operand:DI 1 "pic_32bit_operand" "")
+ 	    (match_operand:SI 1 "pic_32bit_operand" "")
 	      (const_string "lea")
 	   ]
 	   (const_string "imov")))
-- 
1.7.1

